projects/storefrontlib/src/cms-components/product/product-images/product-images.component.ts

```diff
@@ -1,68 +1,76 @@
 import { ChangeDetectionStrategy, Component } from '@angular/core';
 import { Product } from '@spartacus/core';
-import { BehaviorSubject, combineLatest, Observable } from 'rxjs';
-import { filter, map } from 'rxjs/operators';
+import { CarouselItem } from 'projects/storefrontlib/src/shared';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { distinctUntilChanged, filter, map, tap } from 'rxjs/operators';
 import { CurrentProductService } from '../current-product.service';
 
-const WAITING_CLASS = 'is-waiting';
-
 @Component({
   selector: 'cx-product-images',
   templateUrl: './product-images.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class ProductImagesComponent {
-  waiting: HTMLElement;
-
-  product$: Observable<Product> = this.currentProductService
-    .getProduct()
-    .pipe(filter(Boolean));
-
-  private _imageContainer$ = new BehaviorSubject(null);
+  mainMediaContainer = new BehaviorSubject(null);
 
-  imageContainer$ = combineLatest(this.product$, this._imageContainer$).pipe(
-    map(([product, container]) =>
-      container
-        ? container
-        : product.images && product.images.PRIMARY
-        ? product.images.PRIMARY
-        : {}
+  private product$: Observable<
+    Product
+  > = this.currentProductService.getProduct().pipe(
+    filter(Boolean),
+    distinctUntilChanged(),
+    tap((p: Product) =>
+      this.mainMediaContainer.next(p.images ? p.images.PRIMARY : null)
```

Comment:
I think there is a bug here. It might be because of the null but I feel like the media component should handle it 🤷‍♂ . Anyway, for a product with no image we get nothing instead of the "no image" image.

![Screen Shot 2019-06-13 at 10 47 18 AM](https://user-images.githubusercontent.com/10605724/59442741-cc2d3080-8dc8-11e9-9415-9af7f9893b30.png)